core/src/com/unciv/logic/automation/civilization/NextTurnAutomation.kt

```diff
@@ -768,23 +770,78 @@ object NextTurnAutomation {
-                            && it.getDiplomacyManager(civInfo).isRelationshipLevelGT(RelationshipLevel.Neutral)
-                            && !civInfo.getDiplomacyManager(it).hasFlag(DiplomacyFlags.DeclarationOfFriendship)
-                            && !civInfo.getDiplomacyManager(it).hasFlag(DiplomacyFlags.Denunciation)
-                }
-                .sortedByDescending { it.getDiplomacyManager(civInfo).relationshipLevel() }
-        for (civ in civsThatWeCanDeclareFriendshipWith) {
+                .filter { civInfo.diplomacyFunctions.canSignDeclarationOfFriendshipWith(it) }
+                .sortedByDescending { it.getDiplomacyManager(civInfo).relationshipLevel() }.toList()
+        for (otherCiv in civsThatWeCanDeclareFriendshipWith) {
             // Default setting is 5, this will be changed according to different civ.
-            if ((1..10).random() <= 5)
-                civInfo.getDiplomacyManager(civ).signDeclarationOfFriendship()
+            if ((1..10).random() <= 5 && wantsToSignDeclarationOfFrienship(civInfo, otherCiv)) {
+                otherCiv.popupAlerts.add(PopupAlert(AlertType.DeclarationOfFriendship, civInfo.civName))
+            }
         }
     }
 
+    private fun wantsToSignDeclarationOfFrienship(civInfo: Civilization, otherCiv: Civilization): Boolean {
+        // Shortcut, if it is below favorable then don't consider it
+        if (civInfo.getDiplomacyManager(otherCiv).isRelationshipLevelLT(RelationshipLevel.Favorable)) return false
+        
+        val numOfFriends = civInfo.diplomacy.count { it.value.hasFlag(DiplomacyFlags.DeclarationOfFriendship) }
+        val knownCivs = civInfo.getKnownCivs().filter { it.isMajorCiv() && it.isAlive() }.count()
+        val allCivs = civInfo.gameInfo.civilizations.count { it.isMajorCiv() } - 1 // Don't include us
+        val deadCivs = civInfo.gameInfo.civilizations.count { it.isMajorCiv() && !it.isAlive() }
+        val allAliveCivs = allCivs - deadCivs
+        
+        // Motivation should be constant as the number of civs changes
+        var motivation = civInfo.getDiplomacyManager(otherCiv).opinionOfOtherCiv().toInt() - 40
+        
+        if (civInfo.wantsToFocusOn(Victory.Focus.Military)) {
+            // Try to ally with a third of the civs in play
+            // Goes form 0 to -120 as the civ gets more friends, goal is around 1/4 friends max
+            motivation -= ((numOfFriends * 120f) / knownCivs).toInt()
+
+            // Goes from 0 to -50 as more civs die
+            // this is meant to prevent the game from stalemating when a group of friends
+            // conquers all oposition
+            motivation -= deadCivs / allCivs * 50
+        }
+
+        // Wait to declare frienships until more civs
+        // Goes from -30 to 0 when we know 75% of allCivs
+        val civsToKnow = .75f * allAliveCivs
+        motivation -= min(((civsToKnow - knownCivs) / civsToKnow * 30f).toInt(), 0)
```

Comment:
This should *probably* be max(...,  0)
Instead of min/max, please use coerce, for readability - it tends to make these issues trivial

Reply:
Yes, I think I changed it to be positive and then -= without changing the min.